src/meta/Base.ts

```diff
@@ -0,0 +1,23 @@
+import Map from '@dojo/shim/Map';
+import { WidgetMeta, WidgetMetaProperties } from '../interfaces';
+
+export class Base implements WidgetMeta {
+	protected nodes: Map<string, HTMLElement>;
+
+	constructor(properties: WidgetMetaProperties) {
+		this.nodes = properties.nodes;
+		this.invalidate = properties.invalidate;
```

Comment:
I could also just store `invalidate` to `this._invalidate` then write an `invalidate` function that calls `this._invalidate()`